guide/concepts/bins.md

```diff
@@ -0,0 +1,35 @@
+# Bins
+
+Bins, similarly to pipelines, are containers for elements. However, at the same time,
+they can be placed and linked within pipelines. Although bin is a separate
+Membrane entity, it can be perceived as a pipeline within an element.
+Bins can be nested within one another.
+
+There are two main reasons why bins are useful:
+- they enable creating reusable element groups,
+- they allow managing elements inside, for instance by dynamically spawning
+or replacing them as the stream changes.
+
+## Pads
+
+Bin's pads are defined similarly to element's pads and can be linked the same way.
+However, their role is limited to proxy the stream to elements inside (inputs)
+or outside (outputs). To achieve that, each input pad of a bin needs to be linked
+to an output outside and another input inside the bin. Accordingly, each bin's output
+should be linked to output inside and input outside of the bin.
+
+## Bins and the stream
+
+Although the bin passes the stream through its pads, it does not access it directly,
+so that callbacks such as `handle_process` or `handle_event` are not found there.
+This is because the responsibility of the bin is to manage its elements, not to
+process the stream. Whatever the bin needs to know about the stream, it should
+get via notifications from elements inside.
+
+## Bins as black boxes
+
+Bins are designed to take as much responsibility for elements inside as possible ,
+so that pipelines (or parent bins) didn't need to depend on bin's implementation
```

Comment:
They kind of rely on the implementation (it's up to bin's implementation what notifications will be sent). The pipelines shouldn't rely on what elements/bins are inside.